src/MemoryTracing/MemoryTracingIntegrationTest.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <absl/strings/substitute.h>
+#include <absl/synchronization/mutex.h>
+#include <absl/time/clock.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "MemoryTracing/MemoryInfoListener.h"
+#include "MemoryTracing/MemoryInfoProducer.h"
+#include "OrbitBase/Logging.h"
+#include "capture.pb.h"
+
+namespace orbit_memory_tracing {
+namespace {
+
+using orbit_grpc_protos::ProducerCaptureEvent;
+using orbit_grpc_protos::SystemMemoryUsage;
+using orbit_memory_tracing::MemoryInfoListener;
+using orbit_memory_tracing::MemoryInfoProducer;
+
+class BufferMemoryInfoListener : public MemoryInfoListener {
+ public:
+  void OnSystemMemoryUsage(SystemMemoryUsage system_memory_usage) override {
+    ProducerCaptureEvent event;
+    *event.mutable_system_memory_usage() = std::move(system_memory_usage);
+    {
+      absl::MutexLock lock{&events_mutex_};
+      events_.emplace_back(std::move(event));
+    }
+  }
+
+  [[nodiscard]] std::vector<ProducerCaptureEvent> GetAndClearEvents() {
+    absl::MutexLock lock{&events_mutex_};
+    std::vector<ProducerCaptureEvent> events = std::move(events_);
+    events_.clear();
+    return events;
+  }
+
+ private:
+  std::vector<ProducerCaptureEvent> events_;
+  absl::Mutex events_mutex_;
+};
+
+class MemoryTracingIntegrationTestFixture {
+ public:
+  MemoryTracingIntegrationTestFixture(uint64_t memory_sampling_period_ns)
+      : memory_sampling_period_ns_(memory_sampling_period_ns) {}
+
+  void StartTracingAndWaitForTracingLoopStarted() {
+    CHECK(!producer_.has_value());
+    CHECK(!listener_.has_value());
+    producer_.emplace(memory_sampling_period_ns_);
+    listener_.emplace();
+    producer_->SetListener(&*listener_);
+    producer_->Start();
+  }
+
+  [[nodiscard]] std::vector<ProducerCaptureEvent> StopTracingAndGetEvents() {
+    CHECK(producer_.has_value());
+    CHECK(listener_.has_value());
+    producer_->Stop();
+    producer_.reset();
+    std::vector<ProducerCaptureEvent> events = listener_->GetAndClearEvents();
+    listener_.reset();
+    return events;
+  }
+
+ private:
+  uint64_t memory_sampling_period_ns_;
+  std::optional<MemoryInfoProducer> producer_ = std::nullopt;
+  std::optional<BufferMemoryInfoListener> listener_ = std::nullopt;
+};
+
+[[nodiscard]] std::vector<ProducerCaptureEvent> TraceAndGetEvents(
+    MemoryTracingIntegrationTestFixture* fixture) {
+  CHECK(fixture != nullptr);
+
+  fixture->StartTracingAndWaitForTracingLoopStarted();
+
+  constexpr absl::Duration kSleepBeforeStopTracing = absl::Nanoseconds(1000000000);
```

Comment:
use abls::Milliseconds() or Seconds()

Reply:
Changed to use `absl::Miliseconds(1000)`. Thanks for the suggestion!